File: app/src/main/java/com/antyzero/smoksmog/network/NetworkModule.java
package com.antyzero.smoksmog.network;

import android.content.Context;
import android.os.Build;

import java.util.Locale;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import pl.malopolska.smoksmog.SmokSmog;

@Module
public class NetworkModule {

    @Provides
    @Singleton
    public SmokSmog provideSmokSmog(Context context) {
        return new SmokSmog(getLocale(context));
    }

    private Locale getLocale(Context context){
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            return context.getResources().getConfiguration().getLocales().get(0);
        } else {
            //noinspection deprecation
            return context.getResources().getConfiguration().locale;
        }
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/network/NetworkModule.kt
package com.antyzero.smoksmog.network

import android.content.Context
import android.os.Build
import dagger.Module
import dagger.Provides
import pl.malopolska.smoksmog.RestClient
import rx.schedulers.Schedulers
import java.util.*
import javax.inject.Singleton

@Module
open class NetworkModule {

    @Provides
    @Singleton
    internal fun provideSmokSmog(context: Context): RestClient {
        return RestClient.Builder(getLocale(context)).apply {
            scheduler = Schedulers.io()
        }.build()
    }

    private fun getLocale(context: Context): Locale {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            return context.resources.configuration.locales.get(0)
        } else {
            @Suppress("DEPRECATION")
            return context.resources.configuration.locale
        }
    }
}
